### PR TITLE
lua: fix luacheck warnings in src/box/console.lua

### LINKED ISSUE #5032: Fix W212 warnings in src/box/lua/console.lua
## Bug description:

there are a bunch of warnings in `src/box/lua/console.lua` supressed in `.luacheckrc`.
it would be nice to fix them,

## Steps to reproduce:

see steps in `.travis.mk`

### PR DESCRIPTION
**Pull request by Lord-KA**
Closes #5032

### CODE REVIEW

**Comment by ligurio** [comment]
@Lord-KA why you haven't run full ci before requesting code review?

**Comment by Lord-KA** [comment]
My bad, fixed.

**Review by ligurio — CHANGES_REQUESTED** [review]
Gleb, thanks for the patch.
I have some thoughts regarding changes. In fact, changes required for ticket are trivial and there are a number of ways how to do it. Some ways we already discussed verbally.
I'm a bit confused by the method you have chosen to fix - say that variable is used, however it is not used at all in a function's body.
Let me remind the main idea - we need to fix warnings produced by luacheck and don't make code less readable.

**Review by ligurio — APPROVED** [review]
Well, let's  suppress it inline.

LGTM

**Review by Buristan — COMMENTED** [review]
As for me it is more clear to anonymize a variable that isn't used. I don't need to reread function body several times to be sure that this variable is unused. If I want to see what is this unnamed variable, I'll see the function's usage (I already need to do it to understand what is this mystical `storage` or `func` is).

Also, there is no understanding at start what the variable is really unused for functions with more than 2 arguments like `set_param()`.

Moreover, for functions like `help_wrapper()` and `quite()` we have no need to declare this unused argument at all -- just use declaration without arguments. And for `client_handler()` function, we can just omit the last unused `peer` argument.

But this is matter of taste, so feel free to ignore.

**Comment by Totktonada** [comment]
Raw idea: add something like C's `(void)` cast to mark function arguments as unused. Say, introduce the `unused()` function (no-op). Will it work?

**Comment by Lord-KA** [comment]
Once again, I don't think that missing an unused var warning is a big deal. Turning suppression off requires just slightly modifying its comment, you don't have to check for unused vars manually. But generally, I think as you said, the best way to handle it is with a `_` prefix. maybe we can configure it in our `.luacheckrc`?

**Comment by Lord-KA** [comment]
Do I need a doc request for this change?

**Review by Buristan — APPROVED** [review]
Hi, thanks for the fixes!
LGTM, except a single nit below.

Also, you may mention in the commit message that this allows declaration of function (handlers) with an expected standardized interface.

File: .luacheckrc [hunk 1]
```diff
@@ -21,6 +21,7 @@ ignore = {
     "431",
     -- Shadowing an upvalue argument.
     "432",
+    "212/_.*",
```
[the remark this answers is not in the document]
Comment (Lord-KA) [reply .luacheckrc:24]: Thx, missed that.

**Comment by Lord-KA** [comment]
> Hi, thanks for the fixes! LGTM, except a single nit below.
> 
> Also, you may mention in the commit message that this allows declaration of function (handlers) with an expected standardized interface.

Fixed.

**Comment by igormunkin** [comment]
Also backported to 2.10 in scope of b076f7b (2.10.0-183-gb076f7b71).

**Pull request merged**